scripts/filter_advisories.py

```diff
@@ -0,0 +1,163 @@
+import json
+import re
+import sys
+from datetime import datetime
+from typing import Any, Dict, List
+
+import semantic_version
+
+
+def to_snake_case(s: str) -> str:
+    """Convert a string to snake_case."""
+    return re.sub(r"\W|^(?=\d)", "_", s).lower()
+
+
+def read_advisories(filename: str) -> List[Dict[str, Any]]:
+    """Load advisories from a JSON file."""
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+class ValidationError(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+
+def parse_vulnerabilities(
+    advisory: Dict[str, Any],
+    filtered: List[Dict[str, Any]],
+    minimal_version: semantic_version.Version,
+) -> List[Dict[str, Any]]:
+    for vulnerability in advisory.get("vulnerabilities", []):
+        vulnerable_range: str = vulnerability.get("vulnerable_version_range", "")
+
+        if not vulnerable_range:
+            raise ValidationError(
+                f"`vulnerabilities.vulnerable_version_range` field missing for {advisory['ghsa_id']}"
+            )
+
+        try:
+            if "all" in vulnerable_range.lower():
+                filtered.append(advisory)
+                break
+
+            for vuln in vulnerable_range.split(","):
+                parsed_vulnerable_range = re.sub(r"v(\d[.])", r"\1", vuln).replace(
+                    " ", ""
+                )
+
+                range_spec = semantic_version.NpmSpec(parsed_vulnerable_range)
+
+                if minimal_version in range_spec:
+                    filtered.append(advisory)
+                    break
+
+        except ValueError:
+            filtered.append(advisory)
+            break
+
+    return filtered
```

Comment:
I think since you're modifying an array in-place here, it would be clearer to just return `None` here.

Otherwise code like `filtered_advisories = filter_advisories(advisories, minimal_version, last_run_date)` gives the impression you're taking one array as input and returning another one.